File: 프로그래머스/lv1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public Integer[] solution(int []arr) {
        
        Stack<Integer> stackInt = new Stack<>();
        
        stackInt.add(arr[0]); //첫 번째 값 추가
        
        for(int i = 1; i<arr.length; i++){
            
            //제일 최근에 넣은 값을 확인해서 연속된 값인지 아닌지 확인
            if(stackInt.peek() == arr[i]){ 
                continue;
            }else{
                stackInt.add(arr[i]);
            }
        }
        
       Integer[] answer = new Integer[stackInt .size()];
        answer = stackInt.toArray(answer);
        
        return answer;
    }
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.js
class Stack {
  constructor() {
    this._arr = [];
  }
  push(item) {
    this._arr.push(item);
  }
  pop() {
    return this._arr.pop();
  }
  peek() {
    return this._arr[this._arr.length - 1];
  }
length(){
        return this._arr.length;
    }
}

function solution(arr)
{
    
    const stack = new Stack();
    stack.push(arr[0]);
    
    for(let i = 1; i<arr.length; i++){
        if(stack.peek() == arr[i]){
            continue;
        }else{
            stack.push(arr[i]);
        }
    }
    
    let answer = [];
    const index= stack.length();
    for(let i = 0; i<index; i++){
        answer.push(stack.pop());
    }
    
    return answer.reverse();
}